From b2b00b1d7871f7557fe7e8f616fa46a8e5ebd298 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@samba.org>
Date: Wed, 28 May 2014 16:02:15 +0200
Subject: [PATCH] PATCHSET10: s3-libsmbclient: Always initialize globals.

This fixes cases where we dereference NULL pointers of globals which
were not initialized.
---
 source3/libsmb/libsmb_context.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source3/libsmb/libsmb_context.c b/source3/libsmb/libsmb_context.c
index 6c20d65..888c2ef 100644
--- a/source3/libsmb/libsmb_context.c
+++ b/source3/libsmb/libsmb_context.c
@@ -76,7 +76,7 @@ SMBC_module_init(void * punused)
          * defaults ...
          */
 
-        if (!lp_load(get_dyn_CONFIGFILE(), True, False, False, False)) {
+        if (!lp_load(get_dyn_CONFIGFILE(), True, False, False, True)) {
             DEBUG(5, ("Could not load config file: %s\n",
                       get_dyn_CONFIGFILE()));
         } else if (home) {
@@ -89,7 +89,7 @@ SMBC_module_init(void * punused)
             if (asprintf(&conf,
                          "%s/.smb/smb.conf.append",
                          home) > 0) {
-                if (!lp_load(conf, True, False, False, False)) {
+                if (!lp_load(conf, True, False, False, True)) {
                     DEBUG(10,
                           ("Could not append config file: "
                            "%s\n",
-- 
1.9.3

